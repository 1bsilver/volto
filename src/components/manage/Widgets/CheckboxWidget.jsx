/**
 * CheckboxWidget component.
 * @module components/manage/Widgets/CheckboxWidget
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Form, Label, Checkbox } from 'semantic-ui-react';

/**
 * CheckboxWidget component class.
 * @function CheckboxWidget
 * @returns {string} Markup of the component.
 */
const CheckboxWidget = ({ id, title, required, description, error, value, onChange }) =>
  <Form.Field required={required} error={error}>
    <Checkbox
      id={`field-${id}`}
      name={id}
      checked={value}
      label={title}
      onChange={(event, { checked }) => onChange(id, checked)}
    />
    {description && <div className="help">{description}</div>}
    {error && <Label basic color="red" pointing="below">{error}</Label>}
  </Form.Field>;

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
CheckboxWidget.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  required: PropTypes.bool,
  error: PropTypes.string,
  value: PropTypes.bool,
  onChange: PropTypes.func.isRequired,
};

/**
 * Default properties.
 * @property {Object} defaultProps Default properties.
 * @static
 */
CheckboxWidget.defaultProps = {
  description: null,
  required: false,
  error: null,
  value: null,
};

export default CheckboxWidget;
