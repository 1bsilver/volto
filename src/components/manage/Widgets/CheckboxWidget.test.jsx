import React from 'react';
import renderer from 'react-test-renderer';
import CheckboxWidget from './CheckboxWidget';

test('renders a checkbox widget component', () => {
  const component = renderer.create(
    <CheckboxWidget
      id="my-field"
      title="My field"
      onChange={() => {}}
    />,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
