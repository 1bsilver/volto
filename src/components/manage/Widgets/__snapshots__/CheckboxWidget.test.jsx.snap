// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a checkbox widget component 1`] = `
<div
  className="field"
>
  <div
    className="ui checkbox"
    id="field-my-field"
    onChange={[Function]}
    onClick={[Function]}
    onMouseDown={[Function]}
  >
    <input
      checked={null}
      className="hidden"
      name="my-field"
      readOnly={true}
      tabIndex={0}
      type="checkbox"
      value={undefined}
    />
    <label>
      My field
    </label>
  </div>
</div>
`;
